app/models/transfer_factory.rb

```diff
@@ -0,0 +1,79 @@
+class TransferFactory
```

Comment:
This is the first try at encapsulating all this transfer building logic. It's more of a patch to let us move code around for now.